Exiled.Events/Patches/Events/Scp939/PlayingFootstep.cs

```diff
@@ -0,0 +1,68 @@
+// -----------------------------------------------------------------------
+// <copyright file="PlayingFootstep.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp939
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features.Pools;
+    using Exiled.Events.Attributes;
+    using Exiled.Events.EventArgs.Scp939;
+    using Exiled.Events.Handlers;
+
+    using HarmonyLib;
+    using PlayerRoles.FirstPersonControl.Thirdperson;
+    using PlayerRoles.PlayableScps.Scp939.Ripples;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="FootstepRippleTrigger.OnFootstepPlayed" />
+    ///     to add the <see cref="Scp939.PlayingFootstep" /> event.
+    /// </summary>
+    [HarmonyPatch(typeof(FootstepRippleTrigger), nameof(FootstepRippleTrigger.OnFootstepPlayed))]
+    internal static class PlayingFootstep
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label returnLabel = generator.DefineLabel();
+
+            newInstructions.InsertRange(0, new CodeInstruction[]

```

Comment:
It shouldn't be inserted at the beginning. The event invokes when it doesn't even reach to play footstep.

Reply:
> It shouldn't be inserted at the beginning. The event invokes when it doesn't even reach to play footstep.

I've used to insert at the checked part. But there will be errors. I'll try to insert at there again.

Reply:
![image](https://github.com/Exiled-Team/EXILED/assets/45165615/781a9431-a21e-4cfc-b732-78e6ddf5b1db)
The event should be inserted this._syncPlayer. Ideally, it should still fire when invisible check is true, and IsAllowed set to false.

Reply:
> ![image](https://private-user-images.githubusercontent.com/45165615/286778981-781a9431-a21e-4cfc-b732-78e6ddf5b1db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzMDQ3MjcsIm5iZiI6MTcwMTMwNDQyNywicGF0aCI6Ii80NTE2NTYxNS8yODY3Nzg5ODEtNzgxYTk0MzEtYTIxZS00Y2ZjLWI3MzItNzhlNmRkZjViMWRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDAwMzM0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4NjY2ODQwMGMyYTE3ZDI0OTU1MjdhNGUxMGVkY2RkNmJmNDhkNjQ1NDY0ZDQ4ZjJhNmU1YjMyNjcxYTQ2MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-hjYv1KIMzqCTqoWZyD-wK0Aaq5YlV2iXuYBRSZWiyg) The event should be inserted before `this._syncPlayer`. Ideally, it should still fire when invisible check is true, and IsAllowed set to false.


That's what i thought before. I'll change this when i get home asap.

Reply:
Done.

Reply:
> ![image](https://private-user-images.githubusercontent.com/45165615/286780475-be37c358-67c1-49f3-8f38-a915bb62c6fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzMjYxMjYsIm5iZiI6MTcwMTMyNTgyNiwicGF0aCI6Ii80NTE2NTYxNS8yODY3ODA0NzUtYmUzN2MzNTgtNjdjMS00OWYzLThmMzgtYTkxNWJiNjJjNmZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDA2MzAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1OGM3YjcxNjk5MWYzYTI0OTJhMDE3MWUwYWM4YzEyNmM5YmViY2M1NzgwNGJmMmU2MTIzNDJiY2EzNDczZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RC0iVWDdvYPGIDRbj39Y08ZTSRg45WKx2dZPBDlDZhQ) Make sure you handle that label correctly. Otherwise it won't work.

I've done the changes. Now its successfully insert before _syncPlayer.